### ISSUE TITLE
Cannot do automatic silent login at app restart (ADFS)

### ISSUE THREAD

**Issue by nitaeduard** [AzureAD/azure-activedirectory-library-for-objc#1356]
I am struggling for two days with the silent acquire token support. We have the following requirement:
1. app reinstallation and different user logged in
2. app restart should identify the previously logged in user 
3. token expiration and refresh (probably this is the same requirement as 2.)

For this we are having the following initial login setup:
```
    ADAuthenticationContext *authContext = ...;
    [authContext acquireTokenWithResource:resource clientId:clientId redirectUri:redirectURL promptBehavior:AD_PROMPT_ALWAYS completionBlock:^(ADAuthenticationResult *result) {
...
        cachedUserItem = result.tokenCacheItem;
...
    }];

Until this everything is ok, the requirement 1 is done, token is successful retrieved and used later in all the requests. 
The problem appears for the requirements 2 and 3. We have used all the possible user informations from tokenCacheItem but without success, eventually we received errors like AD_ERROR_SERVER_WRONG_USER or AD_ERROR_SERVER_USER_INPUT_NEEDED.

Debugging more the ADAL pod (v2.7.3) implementation we came up with the following simple code, and saw that if we don't pass any userId for the acquireTokenSilentWithResource method, everything works correctly except the situation when there was previously a different user logged in. In this situation AD_ERROR_CACHE_MULTIPLE_USERS is received. 

What userId should be passed to acquireTokenSilentWithResource? We have tried all possibilities (tokenCacheItem.userInformation.userId, tokenCacheItem.userInformation.unp, tokenCacheItem.userInformation.uniqueId) but without success.

```
    ADAuthenticationContext *authContext = ...;
    [authContext acquireTokenSilentWithResource:resource clientId:clientId redirectUri:redirectURL completionBlock:^(ADAuthenticationResult *result) {
        // result == AD_SUCCEEDED

        NSString* userId = result.tokenCacheItem.userInformation.userId;
        // userId = user@domain

        [authContext acquireTokenSilentWithResource:resource clientId:clientId redirectUri:redirectURL userId: userId completionBlock:^(ADAuthenticationResult *result) {
            // result == AD_FAILED
            // because method +ADAuthenticationContext.updateResult generates AD_ERROR_SERVER_WRONG_USER
            // due to result of this method ADUserIdentifier.identifier.matchesInfo
            // - identifier: ADUserIdentifier is like "dGVzdA...="
            // - info: ADUserInformation is "user@domain" (RequiredDisplayableId)
        }];
    }];
```

**Comment by oldalton** [comment]
Hi. Can you please confirm which ADFS version it is? 
Generally, result.tokenCacheItem.userInformation.userId should be used for silent requests. Certain ADFS versions didn't support OIDC, so ADAL saved tokens with an empty user identifier for those.
Is this something that started happening recently or has never worked? 

For further investigation, it would be helpful if you could send [ADAL logs](https://github.com/AzureAD/azure-activedirectory-library-for-objc/wiki/Logging) and [network trace](https://github.com/AzureAD/azure-activedirectory-library-for-objc#network-traces) to [oldalton@microsoft.com](mailto:oldalton@microsoft.com). Please make sure to remove passwords from the network trace prior to sending.

**Comment by oldalton** [comment]
Will be fixed in ADAL 2.7.7 release, PR #1357

**Comment by oldalton** [comment]
This should be now fixed in [ADAL 2.7.7 release](https://github.com/AzureAD/azure-activedirectory-library-for-objc/releases/tag/2.7.7). Please retry and let us know if it's fixed now.

**Comment by nitaeduard** [comment]
Yes, it is fixed in version 2.7.7.
Thank you.

On 23 Dec 2018, at 02:08, Olga Dalton <notifications@github.com<mailto:notifications@github.com>> wrote:


This should be now fixed in ADAL 2.7.7 release<https://github.com/AzureAD/azure-activedirectory-library-for-objc/releases/tag/2.7.7>. Please retry and let us know if it's fixed now.
